packages/expo-barcode-scanner/android/src/main/java/expo/modules/barcodescanner/utils/Helper.java

```diff
@@ -0,0 +1,51 @@
+package expo.modules.barcodescanner.utils;
+
+import android.os.Bundle;
+import android.util.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Helper {
+
+  public static Pair<List<Bundle>,Bundle> getCornerPointsAndBoundingBox(List<Integer> cornerPoints, float density) {
```

Comment:
Please create separate methods:
- `List<Point> scaleCornerPoints(List<Integer> cornerPoints, float scale)`
- `Size getSize(List<Integer> cornerPoints, float scale)`,
- `Point getOrigin(List<Point> cornerPoints)`.

This will require remembering `Point`s and `Size` in `BarCodeScannerResult`, this way it will be the consumer of the `Result` who decides how he would like serialize the result, not the producer. Splitting the function into multiple methods will create more `for`-loops, but also will make this helper more reusable and readable.